aten/src/ATen/native/LinearAlgebraUtils.h

```diff
@@ -317,7 +317,7 @@ static inline void singleCheckErrors(int64_t info, const c10::string_view name,
     } else if (name.find("solve") != name.npos) {
       // solve, linalg_solve, cholesky_solve, etc.
       TORCH_CHECK_LINALG(false, name, batch_string,
-          ": The diagonal element ", info, " is zero, the solve could not be completed because the input matrix is singular.");
```

Comment:
This change hides the information available in `info` hinting the user to the exact problematic place in the matrix.